akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphMergePrioritizedSpec.scala

```diff
@@ -0,0 +1,144 @@
+package akka.stream.scaladsl
+
+import akka.NotUsed
+import akka.stream.testkit.TestSubscriber.ManualProbe
+import akka.stream.{ ClosedShape, Inlet, Outlet }
+import akka.stream.testkit.{ TestSubscriber, TwoStreamsSetup }
+
+class GraphMergePrioritizedSpec extends TwoStreamsSetup {
+  import GraphDSL.Implicits._
+
+  override type Outputs = Int
+
+  override def fixture(b: GraphDSL.Builder[_]): Fixture = new Fixture(b) {
+    val mergePrioritized = b add MergePrioritized[Outputs](Seq(20, 80))
+
+    override def left: Inlet[Outputs] = mergePrioritized.in(0)
+    override def right: Inlet[Outputs] = mergePrioritized.in(1)
+    override def out: Outlet[Outputs] = mergePrioritized.out
+  }
+
+  "merge prioritized" must {
+    commonTests()
+
+    "stream data from all sources" in {
+      val source1 = Source.fromIterator(() ⇒ (1 to 3).iterator)
+      val source2 = Source.fromIterator(() ⇒ (4 to 6).iterator)
+      val source3 = Source.fromIterator(() ⇒ (7 to 9).iterator)
+
+      val priorities = Seq(6, 3, 1)
+
+      val probe = TestSubscriber.manualProbe[Int]()
+
+      threeSourceMerge(source1, source2, source3, priorities, probe).run()
+
+      val subscription = probe.expectSubscription()
+
+      var collected = Seq.empty[Int]
+      for (_ ← 1 to 9) {
+        subscription.request(1)
+        collected :+= probe.expectNext()
+      }
+
+      collected.toSet should be(Set(1, 2, 3, 4, 5, 6, 7, 8, 9))
+      probe.expectComplete()
+    }
+
+    "stream data with priority" in {
+      val elementCount = 10000
+      val source1 = Source.fromIterator(() ⇒ Seq.fill(elementCount)(1).iterator)
+      val source2 = Source.fromIterator(() ⇒ Seq.fill(elementCount)(2).iterator)
+      val source3 = Source.fromIterator(() ⇒ Seq.fill(elementCount)(3).iterator)
+
+      val priorities = Seq(6, 3, 1)
+
+      val probe = TestSubscriber.manualProbe[Int]()
+
+      threeSourceMerge(source1, source2, source3, priorities, probe).run()
+
+      val subscription = probe.expectSubscription()
+
+      var collected = Seq.empty[Int]
+      for (_ ← 1 to elementCount) {
+        subscription.request(1)
+        collected :+= probe.expectNext()
+      }
+
+      val ones = collected.count(_ == 1).toDouble
+      val twos = collected.count(_ == 2).toDouble
+      val threes = collected.count(_ == 3).toDouble
+
+      (ones / twos).round shouldEqual 2
+      (ones / threes).round shouldEqual 6
+      (twos / threes).round shouldEqual 3
```

Comment:
I will increase no of elements in each source to 20k, currently its 10k. Just saw the test failed in CI.